src/vec_int32_ppc.h

```diff
@@ -0,0 +1,649 @@
+#endif
+#else
+//#warning Implememention pre power8
+  vui32_t z= { 0,0,0,0};
+  vui8_t x;
+  x = vec_popcntb ((vui8_t)vra);
+  r = vec_vsum4ubs (x, z);
+#endif
+  return (r);
+}
+#else
+#undef vec_popcntw
+#define vec_popcntw __builtin_vec_vpopcntw
+#endif
+
+/*! \brief byte reverse each word of a vector unsigned int.
+ *
+ *	For each word of the input vector, reverse the order of
+ *	bytes / octets within the word.
+ *
+ *	@param vra a 128-bit vector unsigned int.
+ *	@return a 128-bit vector with the bytes of each word
+ *	reversed.
+ */
+static inline vui32_t
+vec_revbw (vui32_t vra)
+{
+  vui32_t result;
+
+#ifdef _ARCH_PWR9
+#ifndef vec_revb
+  __asm__(
+      "xxbrw %x0,%x1;"
+      : "=wa" (result)
+      : "wa" (vra)
+      : );
+#else
+  result = vec_revb (vra);
+#endif
+#else
+  const vui64_t vconstp =
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+      CONST_VINT64_DW(0x0302010007060504UL, 0x0B0A09080F0E0D0CUL);
+#else
+      CONST_VINT64_DW(0x0C0D0E0F08090A0BUL, 0x0405060700010203UL);
+#endif
+  result = (vui32_t) vec_perm ((vui8_t) vra, (vui8_t) vra, (vui8_t) vconstp);
+#endif
+
+  return (result);
+}
+
+/** \brief Vector Shift left Word Immediate.
+ *
+ *	Vector Shift left Words each element [0-3], 0-31 bits,
+ *	as specified by an immediate value.
+ *	The shift amount is a const unsigned int in the range 0-31.
+ *	A shift count of 0 returns the original value of vra.
+ *	Shift counts greater then 31 bits return zero.
+ *
+ *	@param vra a 128-bit vector treated as a vector unsigned int.
+ *	@param shb Shift amount in the range 0-31.
+ *	@return 128-bit vector unsigned int, shifted left shb bits.
+ */
+static inline vui32_t
+vec_slwi (vui32_t vra, const unsigned int shb)
+{
+  vui32_t lshift;
+  vui32_t result;
+
+  if (shb < 32)
+    {
+      /* Load the shift const in a vector.  The element shifts require
+         a shift amount for each element. For the immediate form the
+         shift constant is splated to all elements of the
+         shift control.  */
+      if (__builtin_constant_p (shb) && (shb < 16))
+	lshift = (vui32_t) vec_splat_s32(shb);
+      else
+	lshift = vec_splats ((unsigned int) shb);
+
+      /* Vector Shift right bytes based on the lower 5-bits of
+         corresponding element of lshift.  */
+      result = vec_vslw (vra, lshift);
+    }
+  else
+    { /* shifts greater then 31 bits return zeros.  */
+      result = vec_xor ((vui32_t) vra, (vui32_t) vra);
+    }
+
+  return (vui32_t) result;
+}
+
+/** \brief Vector Shift Right Algebraic Word Immediate.
+ *
+ *  Vector Shift Right Algebraic Words each element [0-3],
+ *  0-31 bits, as specified by an immediate value.
+ *  The shift amount is a const unsigned int in the range 0-31.
+ *  A shift count of 0 returns the original value of vra.
+ *  Shift counts greater then 31 bits return the sign bit
+ *  propagated to each bit of each element.
+ *
+ *  @param vra a 128-bit vector treated as a vector signed int.
+ *  @param shb Shift amount in the range 0-31.
```

Comment:
nit: do you want `Shift` capitalized? none of the others (line above and line below) have capitalization. (occurs elsewhere as well)